dsa-solutions/lc-solutions/0000-0099/0048-Rotate-Image.md

```diff
@@ -0,0 +1,215 @@
+---
+id: rotate-image
+title: Rotate Image (LeetCode)
+sidebar_label: 0048-RotateImage
+tags:
+    - Array
+    - Math
+    - Matrix
+description: given an 2D matrix which is representing an image, rotate the image by 90 degrees.
+---
+
+## Problem Description
+
+| Problem Statement                                                                                           | Solution Link                                                                                                                               | LeetCode Profile                                   |
+| :----------------------------------------------------------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------ | :------------------------------------------------- |
+| [Rotate Image](https://leetcode.com/problems/rotate-image/description/)                                         | [Rotate Image Leetcode](https://leetcode.com/problems/rotate-image/solutions/) | [Ayush Chaware](https://leetcode.com/ayushchaware08) |
+
+### Problem Description
+- You are given an `n x n` 2D `matrix` representing an image, rotate the image by 90 degrees (clockwise).
+- You have to rotate the image in-place, which means you have to modify the input 2D matrix directly. DO not allocate another 2D matrix and do the rotation.
+
+### Examples
+
+#### Example 1
+- **Input:** `matrix` = [[1,2,3],[4,5,6],[7,8,9]]
+- **Output:**  [[7,4,1],[8,5,2],[9,6,3]]
+
+#### Example 2:
+
+- **Input:** `matrix` = [[5,1,9,11],[2,4,8,10],[13,3,6,7],[15,14,12,16]]
+- **Output:**  [[15,13,2,5],[14,3,4,1],[12,6,8,9],[16,7,10,11]]
+ 
+### Constraints:
+
+- $n == matrix.length == matrix[i].length$
+- $1 <= n <= 20$
+- $-1000 <= matrix[i][j] <= 1000$
+
+### Approach
+
+
+1. **Initialize Variables:** 
+    - `n` The dimension of the matrix (assuming it is an n x n matrix).
+2. **Transpose the Matrix:**
+    - Loop through each element of the matrix and swap elements symmetrically around the diagonal.
+3. **Reverse Each Row:**
+    - Loop through each row and reverse the elements in place.
+
+### Solution Code
+
+#### C++
```

Comment:
can you checkou [Source Code](https://github.com/CodeHarborHub/codeharborhub.github.io/blob/main/dsa-solutions/lc-solutions/0000-0099/0001-two-sum.md?plain=1) for better undestanding our structure

Reply:
Sure

Reply:
Done With updates